src/schema-association-service.ts

```diff
@@ -137,6 +156,10 @@ async function autoDetectSchema(
                 remoteUrl = repo.state.remotes.find(remote => remote.name === remoteName)?.fetchUrl;
                 logger.log(`Found remote URL for ${workspaceFolder.name}: ${remoteUrl}`, 'SchemaDetection');
             }
+            // get remoteUrl for dev branches
```

Comment:
Why was this new branch added?

Reply:
For one of our repos I noticed that we were not getting the remote url when we were not on the main branch because the condition `repo.state.HEAD?.upstream !== undefined` was false.

Reply:
Interesting...HEAD should always be there, unless you're in a detached state. But it doesn't seem to be harming anything, so sure.